core/src/test/java/org/apache/calcite/avatica/util/ArrayImplTest.java

```diff
@@ -164,6 +165,267 @@
+    final ColumnMetaData.Rep rep = ColumnMetaData.Rep.of(PlainJavaObject.class);
+    ColumnMetaData.AvaticaType objectAvaticaType = ColumnMetaData.scalar(
+        Types.OTHER,
+        typeName,
+        rep
+    );
+
+    // This array type name format comes from these two methods:
+    // org.apache.calcite.rel.type.RelDataTypeFactoryImpl.JavaType.generateTypeString()
+    // org.apache.calcite.sql.type.ArraySqlType.generateTypeString()
+    String arrayTypeName = "JavaType(" + PlainJavaObject.class.toString() + ") ARRAY";
+    ArrayType arrayType = ColumnMetaData.array(objectAvaticaType, arrayTypeName, rep);
+
+    ColumnMetaData javaObjectArrayMetaData = MetaImpl.columnMetaData(
+        "PLAIN_JAVA_OBJECT_ARRAY",
+        1,
+        arrayType,
+        true
+    );
+
+    PlainJavaObject pjo1 = new PlainJavaObject(1, "one");
+    PlainJavaObject pjo2 = new PlainJavaObject(2, "two");
+    PlainJavaObject pjo3 = new PlainJavaObject(3, "three");
+    PlainJavaObject pjo4 = new PlainJavaObject(4, "four");
+
+    ArrayImpl.Factory factory = new ArrayFactoryImpl(Unsafe.localCalendar().getTimeZone());
+
+    // Create two arrays for two rows.
+    Array array1 = factory.createArray(objectAvaticaType, Arrays.<Object>asList(pjo1, pjo2));
+    Array array2 = factory.createArray(objectAvaticaType, Arrays.<Object>asList(pjo3, pjo4));
+    List<List<Object>> rows = Arrays.asList(Collections.<Object>singletonList(array1),
+        Collections.<Object>singletonList(array2));
+
+    try (Cursor cursor = new ListIteratorCursor(rows.iterator())) {
+      List<Accessor> accessors = cursor.createAccessors(
+          Collections.singletonList(javaObjectArrayMetaData),
+          Unsafe.localCalendar(),
+          factory
+      );
+      assertEquals(1, accessors.size());
+      Accessor accessor = accessors.get(0);
+
+      assertTrue(cursor.next());
+      Array actualArray = accessor.getArray();
+
+      Object[] arrayData = (Object[]) actualArray.getArray();
+      assertEquals(2, arrayData.length);
+
+      PlainJavaObject actualObject = (PlainJavaObject) arrayData[0];
+      assertEquals(1, actualObject.intProperty);
+      assertEquals("one", actualObject.stringProperty);
+
+      actualObject = (PlainJavaObject) arrayData[1];
+      assertEquals(2, actualObject.intProperty);
+      assertEquals("two", actualObject.stringProperty);
+
+      assertTrue(cursor.next());
+      actualArray = accessor.getArray();
+      arrayData = (Object[]) actualArray.getArray();
+      assertEquals(2, arrayData.length);
+
+      actualObject = (PlainJavaObject) arrayData[0];
+      assertEquals(3, actualObject.intProperty);
+      assertEquals("three", actualObject.stringProperty);
+
+      actualObject = (PlainJavaObject) arrayData[1];
+      assertEquals(4, actualObject.intProperty);
+      assertEquals("four", actualObject.stringProperty);
+    }
+  }
+
+  /**
+   * Test case for when a column is a list of plain Java objects.
+   *
+   * @throws Exception
```

Comment:
My suggestion here was to turn from the current comment to a plain one along this line (since you are not filling the "javadoc" part, it looks like you want just to add a multi-line comment):
```java
  /*
   * Test case for when a column is an array of plain Java objects.
   * This is a common use case when data come from a dynamic schema source.
   */
```